File: C# Examples/01 Fundamentals/MyFirstMacro C#/SolidWorksMacro.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Forms;

using SolidWorks.Interop.sldworks;
using SolidWorks.Interop.swconst;


namespace MyFirstMacro_C_
{
    public partial class SolidWorksMacro
    {
        public void Main()
        {

            string message;
            int swVersion;
            ModelDoc2 swDoc;
            string swTitle;
            swDoc = (ModelDoc2)swApp.ActiveDoc;
            if (swDoc == null)
            {
                return;
            }
            swTitle = swDoc.GetTitle();
            swVersion = swApp.DateCode();
            //message = "Hello SOLIDWORKS " + swVersion.ToString()
            //    + System.Environment.NewLine + "Document: " + swTitle
            //    + System.Environment.NewLine
            //    + "Would you like to close the active document?";
            //message = "Hello SOLIDWORKS " + swVersion.ToString() + "\nDocument: " + swTitle
            //    + "\nWould you like to close the active document?";
            message = string.Format("Hello SOLDIWORKS {0} \nDocument: {1}\nWould you like to close the active document?", swVersion.ToString(), swTitle);
            //show the message
            DialogResult diaRes;
            diaRes = MessageBox.Show(message, "Close document",
            MessageBoxButtons.YesNo);
            if (diaRes == DialogResult.Yes)
            {
                swApp.CloseDoc(swTitle);
            }
            return;

        }

        // The SldWorks swApp variable is pre-assigned for you.
        public SldWorks swApp;

    }
}

File: 01 Fundamentals/MyFirstMacro/SolidWorksMacro.vb
Imports SolidWorks.Interop.sldworks
Imports SolidWorks.Interop.swconst
Imports System.Runtime.InteropServices
Imports System


Partial Class SolidWorksMacro
    Public Sub main()
        Dim message As String
        Dim swVersion As Integer
        Dim swDoc As ModelDoc2
        Dim swTitle As String
        swDoc = swApp.ActiveDoc
        If swDoc Is Nothing Then Exit Sub
        swTitle = swDoc.GetTitle

        swVersion = swApp.DateCode

        'message = "Hello SOLIDWORKS " & swVersion.ToString _
        '  & vbCrLf & "Document: " & swTitle _
        '  & vbCrLf & "Would you like to close the active document?"

        message = String.Format(
            "Hello SOLIDWORKS {0}{1}Active:{2}{1}Would you like to close the active document?",
            swVersion.ToString, vbCrLf, swTitle)
        'Show the message
        Dim response As MsgBoxResult

        response = MsgBox(message, MsgBoxStyle.YesNo)
        If response = MsgBoxResult.Yes Then
            swApp.CloseDoc(swTitle)
        End If

    End Sub
    ''' <summary>
    ''' The SldWorks swApp variable is pre-assigned for you.
    ''' </summary>
    Public swApp As SldWorks
End Class
